app/api/users.py

```diff
@@ -52,30 +36,29 @@ def __init__(self):
 
 	def get(self, id):
 		''' Show User @id '''
-		try:
-			user = next(get_users_by_id(id))
-		except StopIteration:
+		user = User.query.get(id)
+		if not user:
 			abort(404)
 		return { 'user': marshal(user, user_fields) }
 
 	def put(self, id):
 		''' Edit User @id '''
 		args = self.reqparse.parse_args()
-		try:
-			user = next(get_users_by_id(id))
-		except StopIteration:
+		user = User.query.get(id)
+		if not user:
 			abort(404)
-		user['email'] = args.email
+		user.email = args.email
+		db.session.add(user)
```

Comment:
I don't think you need to have the db.session.add line. See: http://stackoverflow.com/questions/11292612/flask-sqlalchemy-how-to-conditionally-insert-or-update-a-row